docs/security.rst

```diff
@@ -0,0 +1,459 @@
+Report a security issue
+=======================
+
+We take security very seriously.
+Your input and feedback on our security is always appreciated.
+
+You can send urgent or sensitive reports directly to security@twistedmatrix.com
+
+You may use our public key (below) to keep your message safe.
```

Comment:
```suggestion
You may use our PGP public key if you wish.
```

Let's not pretend PGP isn't terrible. Perhaps we should drop this statement entirely?

Since @alex seems to be our response team, I took a look at [cryptography's security policy](https://cryptography.io/en/latest/security/#reporting-a-security-issue) which doesn't mention PGP at all, just suggesting use of transport-level encryption.